From 624395f42a8020792fee0e9d9b5c319716f67ee6 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 9 May 2024 20:16:46 +0200
Subject: [PATCH 13/13] usb: gadget: mv_udc: support second clock shared
 between usb/sata

Enable a second clock "reg" during probe if specified.
This should be used for the shared clock between usb and sata.
When not active, system locks up during udc_start.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/usb/gadget/udc/mv_udc_core.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/udc/mv_udc_core.c b/drivers/usb/gadget/udc/mv_udc_core.c
index f1f4f69c0e90..2b1a82c31cb5 100644
--- a/drivers/usb/gadget/udc/mv_udc_core.c
+++ b/drivers/usb/gadget/udc/mv_udc_core.c
@@ -2138,8 +2138,13 @@ static int mv_udc_probe(struct platform_device *pdev)
 	if (!udc)
 		return -ENOMEM;
 
-	/* udc only have one sysclk. */
-	clk = devm_clk_get(&pdev->dev, NULL);
+	/* usb and sata share a general clock */
+	clk = devm_clk_get(&pdev->dev, "reg");
+	if (!IS_ERR(clk))
+		clk_prepare_enable(clk);
+
+	/* udc have one sysclk. */
+	clk = devm_clk_get(&pdev->dev, "core");
 	if (IS_ERR(clk))
 		return PTR_ERR(clk);
 
-- 
2.35.3

